### PR TITLE
Update generated `environment` option for shadow 3.0

### PR DESCRIPTION
**Pull request by stevenengler**
See shadow/shadow#2875.

### CODE REVIEW

**Review by sporksmith — COMMENTED** [review]
I think this is the first change that breaks compatibility with shadow 2.x? @robgjansen do you want to tag a release first?

**Comment by stevenengler** [comment]
> I think this is the first change that breaks compatibility with shadow 2.x? @robgjansen do you want to tag a release first?

If you do make tag a new release, maybe do the tag at 21706fce1. The last few PRs/commits in main were intended for 3.0 compatibility, and I don't think they break 2.x compatibility, but i also didn't test to make sure. So since they don't add anything for 2.x simulations, I think it would be safer to tag the release before those PRs.

File: tornettools/generate.py [hunk 1]
```diff
@@ -265,7 +265,7 @@ def __server(args, network, server):
     if server['is_hs_server']:
         # this is an onion service, so tgen should only listen on localhost so that we catch errors if we accidentally
         # try connecting to the onion service from "exit" clients
-        process["environment"] = "TGENIP=127.0.0.1"
+        process["environment"] = {"TGENIP": "127.0.0.1"}
```
Comment (sporksmith) [inline tornettools/generate.py:268]: Comment to block merge - maybe wait for tornettools to get tagged, though OTOH I guess if this is blocking anything we're already planning to tag  before the current HEAD so :shrug:

**Review by sporksmith — APPROVED** [review]

**Comment by robgjansen** [comment]
New release here: https://github.com/shadow/tornettools/releases/tag/v2.0.0

Bumping version requires a commit+tag, so I did not tag the older commit hash.

File: tornettools/generate.py (hunk 1 above)
Comment (stevenengler) [reply tornettools/generate.py:268]: There has now been a new release tagged.

**Pull request merged**